cmd/kubeadm/app/phases/controlplane/manifests.go

```diff
@@ -193,18 +194,25 @@ func getAPIServerCommand(cfg *kubeadmapi.MasterConfiguration, k8sVersion *versio
 
 // getControllerManagerCommand builds the right controller manager command from the given config object and version
 func getControllerManagerCommand(cfg *kubeadmapi.MasterConfiguration, k8sVersion *version.Version) []string {
+
 	defaultArguments := map[string]string{
 		"address":                          "127.0.0.1",
 		"leader-elect":                     "true",
 		"kubeconfig":                       filepath.Join(kubeadmconstants.KubernetesDir, kubeadmconstants.ControllerManagerKubeConfigFileName),
 		"root-ca-file":                     filepath.Join(cfg.CertificatesDir, kubeadmconstants.CACertName),
 		"service-account-private-key-file": filepath.Join(cfg.CertificatesDir, kubeadmconstants.ServiceAccountPrivateKeyName),
 		"cluster-signing-cert-file":        filepath.Join(cfg.CertificatesDir, kubeadmconstants.CACertName),
-		"cluster-signing-key-file":         filepath.Join(cfg.CertificatesDir, kubeadmconstants.CAKeyName),
 		"use-service-account-credentials":  "true",
 		"controllers":                      "*,bootstrapsigner,tokencleaner",
 	}
 
+	// If using external CA, pass empty string to controller manager instead of ca.key path, so that the csrsigning controller fails to start
+	if certphase.UsingExternalCA(cfg) {
+		defaultArguments["cluster-signing-key-file"] = ""
```

Comment:
for clarity; I'd always set `cluster-signing-key-file` and `cluster-signing-cert-file` in `defaultArguments`
and in this special case; unset both (although one only is technically needed, it's safest to reset both of them